Week-5/assignment1.js

```diff
@@ -0,0 +1,49 @@
+const fetch = require("isomorphic-fetch");
+
+const BASE_API_URL = "https://jsonplaceholder.typicode.com/todos";
+
+const fetchAllToDos = async () => {
+  const res = await fetch(BASE_API_URL);
+  return res.json();
+};
+
+const fetchToDoById = async (toDoId) => {
+  const res = await fetch(`${BASE_API_URL}/${toDoId}`);
+  return res.json();
+};
+
+// GENERATOR FUNCTION IMPLEMENTATION
+async function* getFirstToDoItemFromTheList() {
+  const allToDos = await fetchAllToDos();
+  yield allToDos;
+  const todoItem = await fetchToDoById(allToDos[0].id);
+  return todoItem;
+}
+
+// const iterator = getFirstToDoItemFromTheList();
+// iterator
```

Comment:
@relhanakshay can you invoke getFirstToDoItemFromTheList method and call next() and confirm whether value you are getting as expected or not.